hibernate-core/src/main/java/org/hibernate/query/sqm/internal/SqmUtil.java

```diff
@@ -257,7 +258,11 @@ public static boolean isFkOptimizationAllowed(SqmPath<?> sqmPath) {
 	 * or one that has an explicit on clause predicate.
 	 */
 	public static boolean isFkOptimizationAllowed(SqmPath<?> sqmPath, EntityAssociationMapping associationMapping) {
-		if ( associationMapping.isFkOptimizationAllowed() && sqmPath instanceof SqmJoin<?, ?> sqmJoin ) {
+		if ( associationMapping.isFkOptimizationAllowed()
+			 && sqmPath instanceof SqmJoin<?, ?> sqmJoin
+			 // Allow using the FK columns if the association has a join table
+			 // to avoid joining target tables if necessary
```

Comment:
I would change this comment making a remark about why we don't allow using fk columns when there is no join table, which is the purpose of this change. It may be a little unclear when coming back to this check.

```suggestion
			 // Allow using the FK columns if the association has a join table to avoid rendering
			 // joins when possible, otherwise allow using the target table's mapping
```